### ISSUE TITLE
Analyzers' TaskCanceledExceptions in Error List window

### ISSUE THREAD

**Issue by miloush** [dotnet/roslyn#16518]
**Version Used**: 2.0.0.61313

Not sure how I managed it but I got this in the Error List window.
```
Warning	AD0001
Analyzer 'Microsoft.CodeAnalysis.Xaml.Diagnostics.Analyzers.XamlDocumentDiagnosticAnalyzer' 
threw an exception of type 'System.AggregateException' with message 'One or more errors occurred.'

   > Analyzer 'Microsoft.CodeAnalysis.Xaml.Diagnostics.Analyzers.XamlDocumentDiagnosticAnalyzer'
   threw the following exception: 'System.Threading.Tasks.TaskCanceledException:
   A task was canceled.'.
```
Perhaps cancelled analyzers shouldn't be there?

**Comment by CyrusNajmabadi** [comment]
An aggregate exception indicates a bug.  it means cancellation on a specific token is occurring in a context where cancellation for that token is not expected.  Tagging @mgoertz-msft

**Comment by CyrusNajmabadi** [comment]
Marco, this sort of thing generally happens when someone has captured an incorrect cancellation token (i.e. a token from some previous codepath) and cancels on it later on.

**Comment by mgoertz-msft** [comment]
Is this somehow reproducable? Do you have any further information on the TaskCanceledException like a callstack for example?

I don't think we ever cause a cancellation or store the CancellationToken, we only pass them along. We do hold on to some of the tasks that contain them though but I have never seen this warning before.

**Comment by miloush** [comment]
I might be able to get it again - it was a WPF project and that exception was there pretty much all the time. However, is there a way for me to get the call stack for the TaskCanceledException? If I remember correctly stack of inner exception is not in shown in the Error List window...

**Comment by mgoertz-msft** [comment]
If it's not in there then you would need to attach a debugger from another VS and change the exception settings so that it stops on TaskCanceledExceptions. Once stopped in the debugger you should be able to see exactly who canceled what.

**Comment by CyrusNajmabadi** [comment]
I don't see htis in prism.
